spec/requests/notes_spec.rb

```diff
@@ -0,0 +1,69 @@
+require 'rails_helper'
+# rubocop:disable Metrics/BlockLength
+RSpec.describe 'Notes', type: :request do
+  # rubocop:enable Metrics/BlockLength
+  let(:user) { FactoryGirl.create(:user) }
+  let(:personality) { FactoryGirl.create(:user_personality, user: user) }
+  let(:category) { FactoryGirl.create(:category, user_personality: personality) }
+  let(:transaction) { FactoryGirl.create(:transaction, category: category) }
+  let(:note) { FactoryGirl.create(:note, tranzaction: transaction) }
+
+  before do
+    sign_in user
+  end
+
+  describe 'GET routes tests for notes' do
+    it 'should show new template' do
+      get url_for([:new, personality, category, transaction, :note])
+      expect(response.body).to include 'Create note'
+      expect(response).to have_http_status(200)
+    end
+
+    it 'should show edit template' do
+      get url_for([:edit, personality, category, transaction, note])
+      expect(response.body).to include 'Update note'
+      expect(response).to have_http_status(200)
+    end
+  end
+
+  describe 'positive POST/PATCH/DELETE routes tests for transactions' do
+    it 'should create note' do
+      description = 'Desctiption example'
+      post url_for([personality, category, transaction, Note.new]),
+           params: { note: { description: description } }
+      expect(Note.last.description).to eq(description)
+      expect(response).to have_http_status(302)
```

Comment:
status 302 for positive case? 
How i know 302 status it is for not auth users in devise

Reply:
Yeah, but after creating am redirecting user to url_for([@personality, @category]), so, i can follow_redirect! and expect status code 200 or just expect 302